model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/RecordSimilarityFieldExplainInfo.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 Basis Technology Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.basistech.rosette.apimodel.recordsimilarity;
+
+import lombok.Builder;
+import lombok.Value;
+import lombok.extern.jackson.Jacksonized;
+
+@Builder
+@Jacksonized
+@Value
+public class RecordSimilarityFieldExplainInfo {
+    double weight;
+    double calculatedWeight;
+    double rawScore;
+    double finalScore;
```

Comment:
would any of these ever be null?

Reply:
no, they should always be present given that the object itself is present